pkg/workload/workload_test.go

```diff
@@ -310,3 +311,51 @@ func containersForRequests(requests ...map[corev1.ResourceName]string) []corev1.
 	}
 	return containers
 }
+
+func TestSchedulingTimestamp(t *testing.T) {
+	creationTime := metav1.Now()
+	conditionTime := metav1.NewTime(time.Now().Add(time.Hour))
+	cases := map[string]struct {
+		wl   *kueue.Workload
+		want metav1.Time
+	}{
+		"no condition": {
+			wl: utiltesting.MakeWorkload("name", "ns").
+				Creation(creationTime.Time).
+				Obj(),
+			want: creationTime,
+		},
+		"evicted by preemption": {
+			wl: utiltesting.MakeWorkload("name", "ns").
+				Creation(creationTime.Time).
+				Condition(metav1.Condition{
+					Type:               kueue.WorkloadEvicted,
+					Status:             metav1.ConditionTrue,
+					LastTransitionTime: conditionTime,
+					Reason:             kueue.WorkloadEvictedByPreemption,
+				}).
+				Obj(),
+			want: creationTime,
+		},
+		"evicted by PodsReady timeout": {
+			wl: utiltesting.MakeWorkload("name", "ns").
+				Creation(creationTime.Time).
+				Condition(metav1.Condition{
+					Type:               kueue.WorkloadEvicted,
+					Status:             metav1.ConditionTrue,
```

Comment:
Also test False, which should behave the same as "no condition"

Reply:
That's why I think we shouldn't change the Evicted condition back to False until the job is re-admitted again.

Reply:
I'd prefer to keep it as it is, doing it at the time of admission, will either require to do two SSA updates or make the SSA status updates more complicated (having the admission field manager manage the eviction condition as well).

Reply:
IMO, good API practices and performance comes before the complexity of the management.

Reply:
done

Reply:
added